### ISSUE TITLE
[llvm-diff] False alarm on PHI node of identical modules

### ISSUE THREAD

**Issue by Nuullll** [llvm/llvm-project#56076]
Reproducer:

```bash
$ llvm-diff a.ll a.ll

in function foo:
  in block %exit:
    >   %res = phi double [ 0.000000e+00, %entry ], [ %0, %else ]
    <   %res = phi double [ 0.000000e+00, %entry ], [ %0, %else ]
```

```llvm
; a.ll
define double @foo() {
entry:
  br i1 true, label %else, label %exit

else:
  %0 = extractelement <2 x double> zeroinitializer, i64 0
  br label %exit

exit:
  %res = phi double [ 0.000000e+00, %entry ], [ %0, %else ]
  ret double 0.000000e+00
}
```

llvm version: head c5990d353e144bb490ce98be2e9d116fede978ca

**Comment by Nuullll** [comment]
Working on this: https://reviews.llvm.org/D128024
